engine/schema/src/com/cloud/usage/dao/UsageDaoImpl.java

```diff
@@ -469,4 +478,25 @@ public void removeOldUsageRecords(int days) {
             txn.close();
         }
     }
+
+    @SuppressWarnings("deprecation")
+    public Pair<List<? extends UsageVO>, Integer> getUsageRecordsPendingQuotaAggregation(final long accountId, final long domainId) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        TransactionLegacy.open(TransactionLegacy.USAGE_DB).close();
+        s_logger.debug("getting usage records for account: " + accountId + ", domainId: " + domainId);
+        Filter usageFilter = new Filter(UsageVO.class, "startDate", true, 0L, 10000L);
+        SearchCriteria<UsageVO> sc = createSearchCriteria();
+        if (accountId != -1) {
+            sc.addAnd("accountId", SearchCriteria.Op.EQ, accountId);
+        }
+        if (domainId != -1) {
+            sc.addAnd("domainId", SearchCriteria.Op.EQ, domainId);
+        }
+        sc.addAnd("quotaCalculated", SearchCriteria.Op.NULL);
+        sc.addOr("quotaCalculated", SearchCriteria.Op.EQ, 0);
+        s_logger.debug("Getting usage records" + usageFilter.getOrderBy());
```

Comment:
Debug logging statements should be wrapped in a ``if (s_logger.isDebugEnabled())`` check to avoid the overhead of string concatenation when debug logging is not enabled.